### PR TITLE
Add .editorconfig for automatically setting indentation formatting in…

### PR DESCRIPTION
**Pull request by jackun**
… supported IDEs or text editors

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Default settings:
+# Use 4 spaces as indentation
+[*]
+indent_style = space
+indent_size = 4
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[src/overlay*{cpp,h}]
+indent_size = 3
```
Comment (stephanlachnit) [inline .editorconfig:17]: Are you sure this shouldn't be four?

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+trim_trailing_whitespace = true
```
Comment (stephanlachnit) [inline .editorconfig:18]: Also this whole section is kinda unnecessary since the behavior is already defined in `[*]`.

File: .editorconfig (hunk 1 above)
Comment (jackun) [reply .editorconfig:17]: Most of overlay.* etc are 3 spaces as it was in mesa vulkan overlay.

Comment (stephanlachnit) [reply .editorconfig:17]: Ah whoops I just read `*{cpp,h}`, so nvm.

**Comment by stephanlachnit** [comment]
Btw I would add an exception for Dear ImGui, else it could add newlines in there.

**Comment by jackun** [comment]
> Btw I would add an exception for Dear ImGui, else it could add newlines in there.

It has its own .editorconfig and at least with KDevelop it seems to override MangoHud's settings as it has [root = true](https://stackoverflow.com/a/30539967)

**Comment by stephanlachnit** [comment]
Ah whoops I missed that.

**Pull request merged**